### PR TITLE
Ceph: remove configOverrides from CRD

### PR DESCRIPTION
**Pull request by BlaineEXE**
**Description of your changes:**
We very briefly discussed whether we should remove `configOverrides` from the Ceph CRD. This PR will disable that functionality from the user without removing support in Rook's code. We may choose to take this PR as-is and decide permanently whether to enable the feature in the future, or we may decide to go a step further here and remove support altogether.

**Checklist:**

- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [x] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [x] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test ceph]

### CODE REVIEW

**Comment by BlaineEXE** [comment]
Relevant discussion comments in an unrelated PR: 
- https://github.com/rook/rook/pull/3745#discussion_r320431635
- https://github.com/rook/rook/pull/3745#discussion_r320477411

**Comment by BlaineEXE** [comment]
Can we try to focus on getting this change in before #3745? I think it will be easier to focus on one at a time and less work rebasing for me.

**Comment by travisn** [comment]
@BlaineEXE Yes, reviewing this one now, not sure how i missed it before...

File: Documentation/ceph-configuration.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+title: Configuration
+weight: 3700
+indent: true
+---
+
+# Configuration
+
+For most any Ceph cluster, the user will want to -- and may need to -- change some Ceph
```
Comment (travisn) [inline Documentation/ceph-configuration.md]: nit: should the hyphens not have whitespace between the words? I'm confused by the spacing.

File: Documentation/ceph-configuration.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+[Ceph configuration documentation](https://docs.ceph.com/docs/master/rados/configuration/) before
+making any changes. Changing the settings could result in unhealthy daemons or even data loss if
+used incorrectly.
+
+
+## Required configurations
+Rook and Ceph both strive to make configuration as easy as possible, but there are some
+configuration options which users are well advised to consider for any production cluster.
+
+### Default PG and PGP counts
```
Comment (travisn) [inline Documentation/ceph-configuration.md]: With #3769 we could add the recommendation here to enable the autoscale module. #3769 also makes it the default, assuming we go with that. But if they don't enable it, this section is still very helpful to configure for PGs.

File: Documentation/ceph-configuration.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+`osd_pool_default_pg_num` and `osd_pool_default_pgp_num`. The number of PGs and PGPs can be
+configured on a per-pool basis, but it is highly advised to set default values that are appropriate
+for your Ceph cluster. Appropriate values depend on the number of OSDs the user expects to have
+backing each pool. The Ceph
+[OSD and Pool config docs](https://docs.ceph.com/docs/master/rados/operations/placement-groups/#a-preselection-of-pg-num)
+provide detailed information about how to tune these parameters.
+
+An easier option exists for Rook-Ceph clusters running Ceph Nautilus (v14.2.x) or newer. Nautilus
```
Comment (travisn) [inline Documentation/ceph-configuration.md]: #3769 automates the next two paragraphs

Comment (BlaineEXE) [reply Documentation/ceph-configuration.md]: Is #3769 feature creeping our release?

File: Documentation/ceph-configuration.md (hunk 2 above)
Comment (BlaineEXE) [reply Documentation/ceph-configuration.md]: Is #3769 feature creeping our release?

File: Documentation/ceph-configuration.md (hunk 1 above)
Comment (BlaineEXE) [reply Documentation/ceph-configuration.md]: I think by most US style guides, the em dash isn't separated by spaces, yes.

File: Documentation/ceph-configuration.md (hunk 2 above)
Comment (travisn) [reply Documentation/ceph-configuration.md]: How about we get this PR in first and discuss #3769 separately.

**Comment by BlaineEXE** [comment]
Those seem like reasonable arguments. I'll get this PR's merge  conflicts resolved.

**Pull request merged**